docs: add a docs feedback template issue

alequetzalli opened the pull request:
This is a result of the work mentioned in https://github.com/asyncapi/website/issues/453.

Adding a GH template issue to retrieve feedback. A link for this template would be embedded eventually in our `Feedback card`.

derberg commented:
2 questions from my side:

- do we want it to be a global template for all the repos (this is how it will work if we merge it into `.github` repo. Or rather use it in the `website` repo only for now?
- I think it would be awesome to do the template the way @doc-jones proposed it [here](https://github.com/asyncapi/event-gateway/pull/75) where the template looks like an actual form. Super nifty and so far everyone was very positive about the idea of using this. I even told Doc that it would make sense to roll it out to other issue templates

alequetzalli commented:
> * do we want it to be a global template for all the repos (this is how it will work if we merge it into `.github` repo. Or rather use it in the `website` repo only for now?

I wanted it to be for all repoes, so that is why I chose `.github` over the `website` repo. I want it to be consistent across the board.

> * I think it would be awesome to do the template the way @doc-jones proposed it [here](https://github.com/asyncapi/event-gateway/pull/75) where the template looks like an actual form. Super nifty and so far everyone was very positive about the idea of using this. I even told Doc that it would make sense to roll it out to other issue templates

Thank you, this is a great idea! I will be using this idea in my next draft commit. 😀👍🏽👍🏽

alequetzalli commented on .github/ISSUE_TEMPLATE/DOCS.yml:
```diff
@@ -0,0 +1,38 @@
+name: Docs improvement request
```
@derberg pingo pingo! 🙊😂😂✨✨✨  I have followed your suggestion about using a .yml file instead. 

Please find this PR ready for review again 😀👍🏽👍🏽

doc-jones reviewed (commented):
line 13

doc-jones reviewed (commented):
Looks great other than on tiny nit.

line 13 "make sure to check to if there's already" 
change to "make sure to check if there's already...."

derberg commented:
@doc-jones fyi, GitHub supports super-nifty approach to suggest changes inline, [check it out](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request), it helps a lot

derberg commented:
@alequetzalli please remember that we should create branches in forks, not in the upstream repos. Don't change anything here, just remember for future

derberg commented on .github/ISSUE_TEMPLATE/DOCS.yml:
```diff
@@ -0,0 +1,38 @@
+description: Suggest how we can improve our Docs.
+labels: enhancement
+title: "[enhancement]: "
+labels: ["enhancement", "triage"]
```
I assume that it means that once the issue is created, it will get these labels, right?
I've never seen us using `triage` issue, what would it mean? do we really need it.

don't you want to add some "docs" related labels?

alequetzalli replied:
>I assume that it means that once the issue is created, it will get these labels, right?

Yes! It adds those labels as options for the user to select.


> I've never seen us using `triage` issue, what would it mean? do we really need it?

TBH, I gave a LOT of thought and research today on the question of `do we really need it?`. 

I wondered what `triage` could look like in an OSS implementation, due to the ever-present under-supply of resources on these projects. Is it non-existent? Can it be argued it's the `To-Do`, `In progress`, `PR`, `Done` columns? Or is it examples also such as...
- Collecting and processing data: only collect and keep the data that is most critical to your community. 
- Community Responses: collecting community discussions, feedback, questions, etc

I just found it interesting to think about that... regardless, I think in this particular case for Docs it may be unnecessary, so I want to remove it in next commit fix. 

> don't you want to add some "docs" related labels?
If this question means if I want to add diff labels that are only doc-related, then I agree with you. There are better labels to add here now that I have had more time to think of it. I now realize that what I want is `new docs issue` and `docs bug`(This would have been perfect for corrections like the current pub/sub confusion that was updated by our community in our Streetlights tutorial.). 

I'm also thinking it makes more sense to have 2 Docs templates; one for `new docs issue` and the second template for `docs bug`. I want to add a 2nd template to this PR now for this, will do so in next commit.

derberg replied:
> I wondered what triage could look like in an OSS implementation

We do triage issues, but every maintainer does it probably different and we never thought on "formalizing it" due to the nature fo the project, that there can be repos that are managed by different people with different habits. Just like for example you prefer project board while others do not need it. So tirage happens, I just don't see how label could help here. Not trying to "kill" an idea, but just not sure we add it just like that without further discussion, at least I think so 😅 

> I'm also thinking it makes more sense to have 2 Docs templates 

I have to admit that my first thought was: "2 templates just for docs, seems like too much" but then I thought "yeah, I remember talking to you about a similar topic in the past, we just need to make a clear stand in the community that DOCS are important, and 2 explicit templates follow this rule" 👍🏼

derberg commented on .github/ISSUE_TEMPLATE/DOCS.yml:
```diff
@@ -0,0 +1,38 @@
+name: Docs improvement request
+description: Suggest how we can improve our Docs.
+labels: enhancement
+title: "[enhancement]: "
```
`Enhancement` is very generic. Also let me ask these questions:
- if we have labels, why do we need an additional prefixes in the title?
- if we really really need this prefix, maybe better would be "Docs feedback:" or something. But yeah, first what is it for really.

example, we have some prefixes in spec and it just looks odd and it is nothing that could not be replaced with proper labels really:
![Screenshot 2021-12-06 at 10 11 35](https://user-images.githubusercontent.com/6995927/144819069-1b497318-74c9-4d42-8ed8-5826d23ecc83.png)

derberg replied:
I'm leaving the concept of "title prefix" up to you but just have a look at the below visualization on how will it look like in a repo. I personally think prefix is a redundancy of info we see in labels, but I also get that it might be just me that thinks this way 😄 

![Screenshot 2021-12-07 at 09 14 39](https://user-images.githubusercontent.com/6995927/144991768-320f38a8-d843-49f6-9075-b54d5e5f3c7d.png)

doc-jones commented:
> @doc-jones fyi, GitHub supports super-nifty approach to suggest changes inline, [check it out](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request), it helps a lot

I know.  I was taking a less intrusive approach. I'm sure you would agree that sometimes it's a good way to start with interacting.

derberg commented:
@doc-jones for me most important to know is that you know the feature 😄 tbh I never thought about inline suggestions as "intrusive" but simply easier for the maintainer to apply or discuss further.

alequetzalli commented:
> @alequetzalli please remember that we should create branches in forks, not in the upstream repos. Don't change anything here, just remember for future

Oh dear, I made a mistake... Not sure how I didn't notice I did this PR incorrectly, but thanks for pointing it out!

alequetzalli commented:
Ok welllll... a gazillion mistakes and tests later, here goes the next draft 😂

In this latest commit fix, you will note that I addressed feedback on improving proposed `labels` and `titles`. I also ended up creating **2 doc issue creation templates**: one for `new docs` and the other for filing `doc bug` reports.

derberg commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,46 @@
+  - type: input
+    id: contact
+    attributes:
+      label: Contact Details
```
we are at risk that someone will provide an email address, does it mean we should use it to contact the person. Should we even try to contact someone is a way different that through GitHub comments?

alequetzalli replied:
ok, I am all for removing this contact section then! I don't quite see why to keep it either

derberg commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,46 @@
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Code of Conduct
```
❤️

derberg commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,46 @@
+    id: terms
+    attributes:
+      label: Code of Conduct
+      description: By submitting this issue, you agree to follow our [Code of Conduct](https://example.com)
```
please provide a working link 😛

derberg commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,46 @@
+      label: Describe the bug.
+      description: Is there a bug in one of our code snippets? Did we explain a concept incorrectly? 
+      placeholder: Please share a clear and concise description of what the bug or mistake is.
+      value: "I would like to file an AsyncAPI Docs Bug 🐞 report about..."
```
if we have default values, then user will not see the placeholder text. I suggest you basically extend the placeholder text like this, multiline in yaml is possible:

```
      placeholder: |
        Please share a clear and concise description of what the bug or mistake is.
        
        I would like to file an AsyncAPI Docs Bug 🐞 report about...
``` 

advantage of not having a default value is that users will not bypass a given field. I mean now it is marked as required, but there is already a default value, so they can submit issue without providing certain info. Just sayin'

have a look at how it looks with just placeholder text and look at part where I did not modify anything an there is default value blocking placeholder text:

![Screenshot 2021-12-07 at 09 08 30](https://user-images.githubusercontent.com/6995927/144990784-2371ec99-9b0d-41a1-8761-e54f82df4dff.png)

derberg replied:
we have the same situation in the other template

derberg replied:
maybe we do not need default at all 🤔

alequetzalli replied:
Oh dear, that is def not good then... I will remove the default and keep the placeholder :)

derberg commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,46 @@
+name: Docs Bug 🐞 report
+description: Report mistakes and/or bugs in our Documentation.
+title: "[Docs Bug 🐞 report]: "
+labels: ["docs", "bug"]
```
please remember to ping me before we merge to get `docs` label in all the repos and set it as default for all the next ones we create. I just checked template in repo where I do not have this label and unfortunately template doesn't enforce it's creation

alequetzalli replied:
I am going to follow the hex codes from our new branding, by Missy. :D 

-  docs label (pink #E50E99): `📑 docs`
- docs bug label (yellow #FFD23F): `🐞 docs bug` 

**NOTE: I will update the labels to include emojis in next commit**

*** 

<img width="307" alt="Screen Shot 2021-12-09 at 3 38 41 PM" src="https://user-images.githubusercontent.com/19964402/145492570-c65fbee5-6aaa-48b7-b7bd-60d6ec6d0fa9.png">

derberg reviewed (changes requested):
These templates are lovely 😍 I made just few suggestions

derberg commented:
@alequetzalli 

> Oh dear, I made a mistake... Not sure how I didn't notice I did this PR incorrectly, but thanks for pointing it out! My understanding of how to do it in the GH UI was wrong, I think I got confused w/ their UI at the time I first made this branch 🤦‍♀️

no worries, it happened to every single one of us, like seriously I think every single one of us did it. I guess it has to do something with "lemme learn from my own mistakes" cause I've never seen anyone do it twice 😆 so no worries, just remember 😄

alequetzalli commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,46 @@
+name: Docs Bug 🐞 report
+description: Report mistakes and/or bugs in our Documentation.
```
agreed, consistency really matters... I will change all ref to "AyncAPI Docs"

KhudaDad414 commented:
/help

alequetzalli commented on .github/ISSUE_TEMPLATE/docs-bug.yml:
```diff
@@ -0,0 +1,37 @@
+name: Docs Bug 🐞 report
```
@derberg I removed the `value` from the templates and only left `placeholder` like we were discussing last... does it look ok now? (I'm not sure if preview mode is enough to tell)

alequetzalli replied:
@KhudaDad414 how do you like the docs templates? Lukasz thought you might enjoy reviewing this PR too :)

KhudaDad414 replied:
@alequetzalli thanks.  🙇 it looks great. 💯

derberg reviewed (approved):
Looks perfect from my point of view. I love those and I think we should get all our templates converted into this kind of "forms".

@KhudaDad414 @fmvilas maybe you wanna have a look. In the meantime, I will tweak my script to add new doc-related labels to all repos

KhudaDad414 reviewed (approved):
LGTM 👍

alequetzalli commented:
@derberg can you tag me to let me know when it's ok to merge this PR? 

I will wait to see if Fran/others comment more feedback, but I know I also needed to wait until you had updated labels in repos throughout, so wanted to make sure I knew when it was safe to merge. 😀👍🏽👍🏽

derberg commented:
@alequetzalli sure thing, let's just give Fran a few days to have a look, and then we can assume he is fine and we can merge. I have my script ready, so final change/update is super quick to do

fmvilas reviewed (approved):
🚀 Ship it!

derberg commented:
All done, `docs` and `documentation` labels removed from repos where they were present
New labels added to all repos

![Screenshot 2021-12-15 at 11 14 44](https://user-images.githubusercontent.com/6995927/146167817-81ef482a-d532-4ad4-9107-4408a8435c87.png)

@alequetzalli remember you need to manually update your issues from your project board with proper labels

derberg commented:
/rtm

derberg commented:
I see it there

![Screenshot 2021-12-15 at 11 21 14](https://user-images.githubusercontent.com/6995927/146168773-ee754837-cca2-47e6-8ce0-7fd01b4d09df.png)

alequetzalli commented:
> All done, `docs` and `documentation` labels removed from repos where they were present New labels added to all repos

WOHOOO 🥳  ! Thank you so much for doing this label work w/ me!
